packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -211,40 +210,31 @@ function compareDecodedValue(a: BeaconDecoded, b: BeaconDecoded) {
   return 0;
 }
 
-export function dropInconsistentBeacons(beacons: BeaconDecoded[]) {
+export function chooseMaximalConsistentSet(beacons: BeaconDecoded[]) {
+  // It's either just one beacon data feed or we've eleminated rest of the beacons in the previous step
   if (beacons.length <= 1) return beacons;
 
   const sortedBeacons = [...beacons].sort(compareDecodedValue);
   let maxConsistent: BeaconDecoded[] = [];
 
-  for (let outerIndex = 0; outerIndex < sortedBeacons.length; outerIndex++) {
-    const tmpConsistent: BeaconDecoded[] = [];
-    const outerBeacon = sortedBeacons[outerIndex];
-    const innerBeacons = sortedBeacons.slice(outerIndex + 1);
-
-    // We don't need to continue checking because we won't be able to find a bigger subset anyway
-    if (innerBeacons.length === 0 || innerBeacons.length < maxConsistent.length) break;
-    // * Assume the data point has the smallest value
-    // * Calculate the maximum possible aggregation where the data point deviates by 1% percentage change
-    //   (Aggregation - CurrentDataPoint) / Aggregation = 1%
-    const aggregation = outerBeacon.decodedValue.mul(HUNDRED_PERCENT).div(NINETY_NINE_PERCENT);
-    // * Calculate the maximum value for a data point to contribute to that aggregation
-    //   (MaximumDataPoint - Aggregation) / Aggregation = 1%
-    const maximumDataPoint = aggregation.mul(HUNDRED_AND_ONE_PERCENT).div(HUNDRED_PERCENT);
-    //  * Find how many data points fall between CurrentDataPoint and MaximumDataPoint
-    // We iterate only over the rest of the array, ignoring previous values
-    for (const innerBeacon of innerBeacons) {
-      if (innerBeacon.decodedValue.gt(maximumDataPoint)) break;
-
-      // When first two elements are consistent we need to add the first one as well
-      if (tmpConsistent.length === 0) {
-        tmpConsistent.push(outerBeacon);
+  for (let start = 0; start < sortedBeacons.length; start++) {
+    // We won't be able to find a bigger subset anyway
+    if (sortedBeacons.length - start < maxConsistent.length) break;
+
+    for (let end = start + 1; end < sortedBeacons.length; end++) {
+      const tmpConsistent = sortedBeacons.slice(start, end + 1);
+      const avg = tmpConsistent
+        .reduce((sum, beacon) => sum.add(beacon.decodedValue), ethers.BigNumber.from(0))
+        .div(ethers.BigNumber.from(tmpConsistent.length));
+      const maxDeviation = avg.mul(ONE_PERCENT).div(HUNDRED_PERCENT);
+      if (
```

Comment:
add a comment that it "If boundery values pass the deviation check all others have to pass it as well" (or some better worded comment like this).